src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -181,10 +188,10 @@ public void parseButtonClicked() {
             return;
         }
 
-        BackgroundTask.wrap(() -> new DefaultTexParser().parse(fileList))
+        BackgroundTask.wrap(() -> new TexBibEntriesResolver(databaseContext.getDatabase(), preferencesService.getImportFormatPreferences(), fileMonitor).resolve(new DefaultTexParser().parse(fileList)))
```

Comment:
this is a bit hard to understand, I would introduce a variable isntea of directly using the resolve